packages/peregrine/lib/state/Cart/state.js

```diff
@@ -0,0 +1,165 @@
+import { useCallback, useMemo, useReducer } from 'react';
+import * as actions from './actions';
+import withLogger from '../../util/withLogger';
+
+const initialState = {
+    cartId: null,
+    details: {},
+    paymentMethods: [],
+    totals: {}
+};
+
+const reducer = (state, { payload, type }) => {
+    switch (type) {
+        case 'reset cart': {
+            return {
+                ...initialState
+            };
+        }
+        case 'set cart id': {
+            return {
+                ...state,
+                cartId: payload
+            };
+        }
+        case 'set details': {
+            return {
+                ...state,
+                details: {
+                    ...payload
+                }
+            };
+        }
+        case 'set payment methods': {
+            return {
+                ...state,
+                paymentMethods: [...payload]
+            };
+        }
+        case 'set totals': {
+            return {
+                ...state,
+                totals: {
+                    ...payload
+                }
+            };
+        }
+        default: {
+            return state;
+        }
+    }
+};
+
+const wrappedReducer = withLogger(reducer);
+export const useCartState = () => {
+    const [state, dispatch] = useReducer(wrappedReducer, initialState);
+    const { cartId } = state;
+
+    const createCart = useCallback(
+        async isSignedIn => {
+            if (cartId) {
+                return cartId;
```

Comment:
Will this actually update the `cartId` across components? ie If componentA calls `createCart` which should dispatch to the state, what happens when componentB calls `createCart` later?